simpleclient/src/test/java/io/prometheus/client/DisableCreatedTest.java

```diff
@@ -0,0 +1,92 @@
+package io.prometheus.client;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
+
+import java.util.Collections;
+import java.util.List;
+
+public class DisableCreatedTest {
+
+    private CollectorRegistry registry;
+
+    @Before
+    public void setUp() {
+        registry = new CollectorRegistry(true);
+    }
+
+    @Test
+    public void testDisableCreatedAll() throws Exception {
+        EnvironmentVariables env = new EnvironmentVariables(Collector.DISABLE_CREATED_SERIES, "true");
+        Counter counter1 = Counter.build()
+                .name("counter1")
+                .help("test counter 1")
+                .labelNames("path")
+                .register(registry);
+        counter1.labels("/hello").inc();
+        counter1.labels("/goodbye").inc();
+        Counter counter2 = Counter.build()
+                .name("counter2")
+                .help("test counter 2")
+                .register(registry);
+        counter2.inc();
+
+        Histogram histogram = Histogram.build()
+                .name("test_histogram")
+                .help("test histogram")
+                .create();
+        histogram.observe(100);
+        histogram.observe(200);
+        registry.register(histogram);
+
+        Summary noLabels = Summary.build().name("test_summary").help("test summary").register(registry);
+        noLabels.observe(2);
+
+        List<Collector.MetricFamilySamples> mfsList;
+        try {
+            env.setup();

```

Comment:
If you make the flag `final`, you'll need to call `env.setup()` before initializing the metrics.

Reply:
How can I do that?

Reply:
That was referring to
> If you make it final but still want to test it, you should not make it static.

There are a few other options how to write a test though.

Reply:
As noted above - some guidance/example on how to use the framework would be appreciated.

Reply:
Have a look at `JavaVersionsIT`. I guess for your test you won't need different Java versions, but apart from that it should be similar.